### ISSUE TITLE
4.6.49 lich and updated dr-scripts are not working together

### ISSUE THREAD

**Issue by Sarvatt** [rpherbig/dr-scripts#3175]
Just got a report that someone not using the lich fork was running into no scripts working and switching to the fork fixed it on ruby 2.5.1, there may be a logic error in https://github.com/rpherbig/dr-scripts/commit/78e3d03a3b09e79b3abd8ac801ca6eb267e5ab4e#diff-9b7b8b61093358062f1ab4ac37b348de , $MODERN_LICH should be true on 4.6.49 also where trust was removed for newer ruby also? stock lich doesn't work out of the box unless someone is using the fork right now.

**Comment by rcuhljr** [comment]
Is it time to kill support for non fork with dependency?

**Comment by Sarvatt** [comment]
i submitted that as https://github.com/rpherbig/dr-scripts/pull/2929 but got pushback :)

**Comment by rcuhljr** [comment]
But if we don't go that way, I think `LICH_VERSION =~ /^(\d+)\.(\d+)\./` should have an f in it for fork. something like `LICH_VERSION =~ /^(\d+)\.(\d+)\\d+f./`

**Comment by rcuhljr** [comment]
>I haven't given this a lot of thought, so I could easily be convinced to do something else (I don't have a strong preference right now).

That's some vicious push back :P

**Comment by Sarvatt** [comment]
not caring enough to convince 💃

**Comment by Sarvatt** [comment]
it could have been implemented better, i am the ruby noob who was just trying to fix https://github.com/rpherbig/dr-scripts/issues/2000 getting in over my head.

**Comment by jandersson** [comment]
> But if we don't go that way, I think `LICH_VERSION =~ /^(\d+)\.(\d+)\./` should have an f in it for fork. something like `LICH_VERSION =~ /^(\d+)\.(\d+)\.\d+f./`

I initially had the check as  `LICH_VERSION =~ /^(\d+)\.(\d+)\.\d+.*(f)./` then removed it because I did not realize there was a problem.

**Comment by rpherbig** [comment]
I'm swamped, can someone ELI5 this for me?

**Comment by jandersson** [comment]
This should be fixed by #3180 and #3179  . The issue was not checking non-fork lich version numbers correctly. The assumption was that everyone was on the fork, which was not a good assumption

**Comment by Sarvatt** [comment]
thanks Jonas!
